src/wp-includes/class-wp-block.php

```diff
@@ -148,26 +175,40 @@ public function __construct( $block, $available_context = array(), $registry = n
 			}
 		}
 
-		if ( ! empty( $block['innerBlocks'] ) ) {
-			$child_context = $this->available_context;
-
+		if ( ! empty( $this->parsed_block['innerBlocks'] ) ) {
 			if ( ! empty( $this->block_type->provides_context ) ) {
 				foreach ( $this->block_type->provides_context as $context_name => $attribute_name ) {
 					if ( array_key_exists( $attribute_name, $this->attributes ) ) {
-						$child_context[ $context_name ] = $this->attributes[ $attribute_name ];
+						$this->available_context[ $context_name ] = $this->attributes[ $attribute_name ];
 					}
 				}
 			}
+		}
+	}
```

Comment:
@mukeshpanchal27 In regards to https://github.com/WordPress/wordpress-develop/pull/7522#discussion_r1801583320, can you please explain why you changed this code in this way, instead of moving it to the other method? Why is it important that this remains in this method?

I am confused to see this now sets `$this->available_context` when until yesterday it didn't. This clause defines the child context, i.e. for inner blocks, as far as I can tell, so I don't think it should be changed to modify _the current block's_ context?

I would still think this clause should be left as it was and be moved to `refresh_parsed_block_dependents()`, since it is not dependent on `$this->context`.

Reply:
I read that only the parts related to `$this->parsed_block` need to be moved, while the context-related changes should remain as they are. So, I only moved the `$this->parsed_block` part and left the context unchanged.

In https://github.com/WordPress/wordpress-develop/pull/7522/commits/c7c54180a55546cf98243e9dd7040274c2f4cbb4 i update it.